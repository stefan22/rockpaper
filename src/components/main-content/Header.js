import React from 'react';
import GameBoard from './game-board/GameBoard';
import CurrentGame from './CurrentGame';
import '../../scss/header.scss';

const Header = (props) => {
  let {wins,losses,draws,games,turn,currentGameResult} = props;
  return (
    <section className='top-score'>
      <div className='container'>
        <div className="row">
          <div className='col-12'>
            <div className='container'>
              <CurrentGame
                currentGameResult={currentGameResult}
              />
              <GameBoard
                wins={wins}
                losses={losses}
                draws={draws}
                games={games}
                turn={turn}
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  );

};

export default Header;
