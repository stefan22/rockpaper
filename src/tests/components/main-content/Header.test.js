import React from 'react';
import {shallow} from 'enzyme';
import Header from '../../../components/main-content/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
