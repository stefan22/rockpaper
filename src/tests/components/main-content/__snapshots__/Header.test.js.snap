// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-12"
          >
            <div
              className="container"
            >
              <CurrentGame />
              <GameBoard />
            </div>
          </div>
        </div>
      </div>,
      "className": "top-score",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-12"
          >
            <div
              className="container"
            >
              <CurrentGame />
              <GameBoard />
            </div>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-12"
          >
            <div
              className="container"
            >
              <CurrentGame />
              <GameBoard />
            </div>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <CurrentGame />
              <GameBoard />
            </div>,
            "className": "col-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <CurrentGame />,
                <GameBoard />,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currentGameResult": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "draws": undefined,
                  "games": undefined,
                  "losses": undefined,
                  "turn": undefined,
                  "wins": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-12"
            >
              <div
                className="container"
              >
                <CurrentGame />
                <GameBoard />
              </div>
            </div>
          </div>
        </div>,
        "className": "top-score",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-12"
            >
              <div
                className="container"
              >
                <CurrentGame />
                <GameBoard />
              </div>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-12"
            >
              <div
                className="container"
              >
                <CurrentGame />
                <GameBoard />
              </div>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="container"
              >
                <CurrentGame />
                <GameBoard />
              </div>,
              "className": "col-12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <CurrentGame />,
                  <GameBoard />,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currentGameResult": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "draws": undefined,
                    "games": undefined,
                    "losses": undefined,
                    "turn": undefined,
                    "wins": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
